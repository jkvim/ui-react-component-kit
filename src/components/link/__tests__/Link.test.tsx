import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Link } from '../';
import mock = jest.mock;

describe('Link', () => {

  const mockCallback = jest.fn();

  describe('props', () => {

    const wrapper = mount(
      <Link
        id="id"
        title="title"
        url="http://google.com.au"
        isExternal={true}
        onClick={mockCallback}
        className="class"
      />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `title` in props', () => {
      expect(wrapper.props().title).toBe('title');
    });

    it('should pass `url` in props', () => {
      expect(wrapper.props().url).toBe('http://google.com.au');
    });

    it('should pass `isExternal` in props', () => {
      expect(wrapper.props().isExternal).toBe(true);
    });

    it('should pass `onClick` in props', () => {
      expect(wrapper.props().onClick).toBe(mockCallback);
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<Link id="id" title="title" url="http://google.com" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot internal', () => {
    const wrapper = mount(<Link id="id" title="title" url="inner" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot external', () => {
    const wrapper = mount(<Link id="id" title="title" url="http://google.com" isExternal={true} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('callback', () => {
    expect(mockCallback).not.toBeCalled();
    const wrapper = mount(
      <Link id="id" title="title" url="http://google.com" onClick={mockCallback} isExternal={true} />);
    wrapper.simulate('click');
    expect(mockCallback).toBeCalled();
  });
});
