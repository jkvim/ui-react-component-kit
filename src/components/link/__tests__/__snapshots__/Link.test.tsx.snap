// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link props should match snapshot 1`] = `
<Link
  className="class"
  id="id"
  isExternal={true}
  onClick={[MockFunction]}
  title="title"
  url="http://google.com.au"
>
  <styled.a
    className="class"
    href="http://google.com.au"
    id="id"
    onClick={[MockFunction]}
    target="_blank"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <a
      className="class sc-ifAKCX beoLJX"
      href="http://google.com.au"
      id="id"
      onClick={[MockFunction]}
      target="_blank"
    >
      title
      <styled.span>
        <span
          className="sc-EHOje jGgJXl"
        >
          <styled.span>
            <span
              className="sc-bdVaJa hjmIub"
            >
              Link opens in new window
            </span>
          </styled.span>
        </span>
      </styled.span>
    </a>
  </styled.a>
</Link>
`;

exports[`Link snapshot basic 1`] = `
<Link
  id="id"
  title="title"
  url="http://google.com"
>
  <styled.a
    href="http://google.com"
    id="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <a
      className="sc-ifAKCX beoLJX"
      href="http://google.com"
      id="id"
    >
      title
    </a>
  </styled.a>
</Link>
`;

exports[`Link snapshot external 1`] = `
<Link
  id="id"
  isExternal={true}
  title="title"
  url="http://google.com"
>
  <styled.a
    href="http://google.com"
    id="id"
    target="_blank"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <a
      className="sc-ifAKCX beoLJX"
      href="http://google.com"
      id="id"
      target="_blank"
    >
      title
      <styled.span>
        <span
          className="sc-EHOje jGgJXl"
        >
          <styled.span>
            <span
              className="sc-bdVaJa hjmIub"
            >
              Link opens in new window
            </span>
          </styled.span>
        </span>
      </styled.span>
    </a>
  </styled.a>
</Link>
`;

exports[`Link snapshot internal 1`] = `
<Link
  id="id"
  title="title"
  url="inner"
>
  <styled.a
    href="inner"
    id="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <a
      className="sc-ifAKCX beoLJX"
      href="inner"
      id="id"
    >
      title
    </a>
  </styled.a>
</Link>
`;
